### PR TITLE
Add /tmp to staging

### LINKED ISSUE #793: v4.0.0-rc4 error with permission denied in /tmp
Attempted to upgrade from v4.0.0-rc3 to v4.0.0-rc4 today.
Immediately receive an error about "permission denied" when creating "/tmp/git-sync.gitconfig.RANDOM_NUMBERS".
When running as a sidecar container (with grafana), the user is NOT `git-sync` but rather `472` due to `securityContext` settings in the helm chart (the `grafana` user).

rc3 permissions (note sticky bit on /tmp):
```
$ ls -al /tmp
total 8
drwxrwxrwt 2 root root 4096 Aug  3  2017 .
drwxr-xr-x 1 root root 4096 Jul 31 14:26 ..
$ ls -al /git
total 8
drwxrwsr-x 2 git-sync git-sync 4096 Jul  7 03:12 .
drwxr-xr-x 1 root     root     4096 Jul 31 14:33 ..
```

rc4 permissions (/tmp owned by git-sync user):
```
$ ls -al /tmp
total 8
drwxr-xr-x 2 git-sync git-sync 4096 Jul 30 00:06 .
drwxr-xr-x 1 root     root     4096 Jul 31 14:27 ..
$ ls -al /git
total 8
drwxrwsr-x 2 git-sync git-sync 4096 Jul 28 19:51 .
drwxr-xr-x 1 root     root     4096 Jul 31 14:42 ..
```

I'm planning to work-around this by messing with security contexts, but wanted to make you aware.

### PR DESCRIPTION
**Pull request by thockin**
Fix #793

### CODE REVIEW

File: Dockerfile.in [hunk 1]
```diff
@@ -68,8 +68,7 @@ RUN /stage_binaries.sh -o {ARG_STAGING} \
 	-f /etc/nsswitch.conf \
 	-f /etc/os-release \
 	-f /etc/passwd \
-	-f /etc/shadow
-RUN ln -sf /bin/dash {ARG_STAGING}/bin/sh
```
Comment (justinsb) [inline Dockerfile.in:72]: Removed deliberately or misclick?

**Comment by justinsb** [comment]
I believe this will copy the permissions for `/tmp` (which are particularly important), because we use `cp -a`

LGTM though I wasn't sure whether you deliberately removed the dash symlink, so will add a hold but feel free to remove hold if deliberate.

/approve
/lgtm
/hold

**Comment by thockin** [comment]
/etc/shadow is not needed

The symlink to dash is already covered by the dash package (redundant)

This *does* retain /tmp permissions:

```
$ make container
<...lots ot output...>
container: gcr.io/k8s-staging-git-sync/git-sync:v4.0.0-rc4-1-g7a72116__linux_amd64

$ docker run -ti --entrypoint "" gcr.io/k8s-staging-git-sync/git-sync:v4.0.0-rc4-1-g7a72116__linux_amd64 ls -ld /tmp
drwxrwxrwt 2 root root 4096 Jul 31 15:18 /tmp
```

**Pull request merged**